### PR TITLE
Laboratory work #1, Zhilina Polina - 21FPL2 

### PR DESCRIPTION
**Pull request by polyamba**

### CODE REVIEW

**Review by artyomtugaryov — COMMENTED** [review]
Hello, @polyamba! As a reminder, the deadline for the first lab is September 30th and you have less than 2 weeks to complete it. Please hurry to be on time.

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -8,32 +8,35 @@
...
     text (str): Original text
-
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
-
     In case of corrupt input arguments, None is returned
     """
-    pass
 
+    clean1 = text.lower().replace(':', '').replace('!', '').replace('...', '').replace('!..', '').replace('?..', '').replace('.', '').replace(',', '').replace('?', '').replace(';', '').replace('-', '')
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:18]: try to use `string.punctuation` to replace all punctuation

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -8,32 +8,35 @@
...
-    stop_words (List[str]: Tokens to exclude
 
-    Returns:
-    List[str]: Token sequence that does not include stop words
 
-    In case of corrupt input arguments, None is returned
+def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
-    pass
+        Excludes stop words from the token sequence
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:27]: revert this cahne^ please

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -157,3 +160,4 @@ def extract_significant_words(chi_values: dict[str, float], alpha: float) -> Opt
     In case of corrupt input arguments, None is returned
     """
     pass
+
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:163]: remove this change

File: lab_1_keywords_tfidf/start.py [hunk 4]
```diff
@@ -31,6 +31,30 @@
     with open(CORPUS_FREQ_PATH, 'r', encoding='utf-8') as file:
         corpus_freqs = json.load(file)
 
+
+    def clean_and_tokenize(text: str) -> Optional[list[str]]:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/start.py:35]: import these functions from the main.py

File: lab_1_keywords_tfidf/main.py [hunk 5]
```diff
@@ -99,12 +138,21 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
+    if not(isinstance(term_freq, dict) and term_freq and isinstance(idf, dict)
+           and all(isinstance(k, str) for k in idf.keys()) and all(isinstance(v, float) for v in idf.values())):
+        return None
+    for key, value in term_freq.items():
+        if not (isinstance(key, str) and isinstance(value, float)):
+            return None
+        tfidf_d = {}
+        for word in term_freq:
+            if word not in idf.keys():
+                idf[word] = math.log(47 / 1)
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:150]: why `/ 1` ??

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Revert every docstring

File: lab_1_keywords_tfidf/main.py (hunk 5 above)
Comment (polyamba) [reply lab_1_keywords_tfidf/main.py:150]: And I want to ask about "Revert every docstring", should I add back strings with pass or what strings? I think other docstrings are at their place or not?

Comment (artyomtugaryov) [reply lab_1_keywords_tfidf/main.py:152]: ask about revert in comments about docs

Comment (artyomtugaryov) [reply lab_1_keywords_tfidf/main.py:152]: Why don't you do 0+1?

**Pull request closed without merging**